res/controllers/Denon-MC7000-scripts.js

```diff
@@ -212,13 +211,10 @@ MC7000.init = function() {
     }
 
     // send Controller Status SysEx message delayed to avoid conflicts with Softtakeover
-    engine.beginTimer(2000, MC7000.delayedSysEx, true);
-};
-
-// SysEx message to receive all knob and fader positions
-MC7000.delayedSysEx = function() {
-    var ControllerStatusSysex = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7];
-    midi.sendSysexMsg(ControllerStatusSysex, ControllerStatusSysex.length);
+    engine.beginTimer(2000, function() {
```

Comment:
I had unreliable behavior when introducing soft takeover for the pitch faders. Sometimes the position was acknowledged, sometimes not, when both, soft takeover and the Sysex where sent "same" time. 

I cannot remember exactly but think in the forum there was a little discussion and maybe it was Be suggesting and delay the Sysex. This works very reliable.

Just made this Syntax change while I learned more about the one line timer as introduced for SLIP modes

Reply:
Mhmm, thats pretty weird and probably rather a symptom of using softTakeover wrong, but its fine as is. Thanks for the clarification.

Reply:
Well, Josep pointed me to other mappings where that was solved also by delaying. If you have any better suggestion, then please let me know. Always interested for improvements

Reply:
Well personally my approach would be to enable softtakeover after synchronizing the state with the controller. I'm not familiar enough with the mapping code to judge how feasible this is though...

Reply:
> Well personally my approach would be to enable softtakeover after synchronizing the state with the controller.

I didn't remember why I made it this way so just turned it around and found out quickly why. Let me try to explain... 

In principal it is logical to let the status be called first and then apply soft takeover. If the pitch fader is **not** at zero position then this works great. If the slider is in zero position and I move it away from it then the software doesn't recognize the movement and stays at original tempo, so I need to get back through zero and only then it works. 

In case the soft takeover is applied after the status request then regardless of where the slider is positioned the track loads with its original tempo. One have to be pick up the track tempo at zero again but it is much more intuitive then the previous scenario, where I have to go from 0 to value back to 0 and then adjust tempo.

Hope this was not too complicated and you could understand the reason.

Cheers!

Reply:
Mhmm that is indeed a strange phenomenon I have trouble to wrap my head around. My guess is that the mapping still employs softtakeover incorrectly but don't really have time to look into it.
Anyway which way, the current state seems to be fine (I guess you haven't received any complaints either from people using this mapping?) so leaving as-is if fine.

Reply:
Hello @Swiftb0y - for some reason I don't see any trouble anymore to keep it like your suggestion. First get the status and then apply soft takeover. My last commit also shows this. Cheers!